language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/RemoveImportCodeAction.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.codeaction.providers.imports;
+
+import io.ballerina.compiler.syntax.tree.ImportDeclarationNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.codeaction.CodeActionNodeValidator;
+import org.ballerinalang.langserver.codeaction.providers.AbstractCodeActionProvider;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.spi.DiagBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Code Action to remove an unused/ re-declared module import.
+ *
+ * @since 2201.1.1
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class RemoveImportCodeAction extends AbstractCodeActionProvider {
+    public static final String NAME = "remove import";
+    public static final Set<String> DIAGNOSTIC_CODES = Set.of("BCE2002", "BCE2004");
+
+    @Override
+    public boolean validate(Diagnostic diagnostic, DiagBasedPositionDetails positionDetails,
+                            CodeActionContext context) {
+        return DIAGNOSTIC_CODES.contains(diagnostic.diagnosticInfo().code()) 
+                && CodeActionNodeValidator.validate(context.nodeAtCursor());
+    }
+
+    @Override
+    public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
+                                                    DiagBasedPositionDetails positionDetails, 
+                                                    CodeActionContext context) {
+        ImportDeclarationNode importDeclNode = positionDetails.matchedNode().kind() == SyntaxKind.IMPORT_DECLARATION 
```

Comment:
@malinthar  Usually the behaviour of the diagnostic is this. But this might break in the case of change in diagnostic behaviour. Should we consider that or is there a scenario which can break this except that?

Reply:
To be on the safe side we can simply introduce a loop and find the import declaration node.

Reply:
+1 to have a strict check since we cannot trust the behavior of the parser/diagnostics.